### PR TITLE
Feature/calendar microcopy 

### PR DESCRIPTION
**Pull request by LindsayYoung**
Almost ready for review, I need to confirm report districts are working with real data.

### CODE REVIEW

**Comment by LindsayYoung** [comment]
The results look good to me. I am going to add electioneering communication reporting periods with states as a separate PR.

**Comment by LindsayYoung** [comment]
closes #1649

**Comment by emileighoutlaw** [comment]
I want to look at this, too! Making this note here for myself 😄

File: data/functions/calendar.sql [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+create or replace function generate_report_description(office_sought text, report_type text, rpt_tp_desc text, contest text[])
+returns text as $$
+    begin
+        return case
+            when rpt_tp_desc is null and array_length(contest, 1) > 3 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Report Multi-state Due Today'
```
Comment (emileighoutlaw) [inline data/functions/calendar.sql]: Would it be more human readable if this said `Report (for Multiple States) is Due Today` or `Report is Due Today in Multiple States`

File: data/functions/calendar.sql [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                ], ' ')
+            when rpt_tp_desc is null and array_length(contest, 1) = 0 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/functions/calendar.sql]: Can we add `is` here? So `Report is Due Today` ?

File: data/functions/calendar.sql [hunk 3]
```diff
@@ -178,8 +178,7 @@ returns text as $$
                 array[
                     expand_office_description(office_sought),
                     report_type,
-                    'Due Today',
-                    'States:',
+                    'Due Today. States:',
```
Comment (emileighoutlaw) [inline data/functions/calendar.sql]: How would you feel about making this say `is Due Today in the Following States:` ?

**Comment by emileighoutlaw** [comment]
Hey there! These are awesome updates Lindsay. I made a couple notes of things I'm interested in changing. If you think that look good to you, I'm happy to hope onto your branch and make them myself, so you don't have to look at all that microcopy again.

File: data/sql_updates/omnibus_dates.sql [hunk 4]
```diff
@@ -68,27 +69,29 @@ returns text as $$
                 array[
                     expand_office_description(office_sought),
                     report_type,
-                    'Report Multi-state'
+                    'Report Multi-state due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 5]
```diff
@@ -68,27 +69,29 @@ returns text as $$
...
                 ], ' ')
             when rpt_tp_desc is null then
                 array_to_string(
                 array[
                     array_to_string(election_state, ', '),
                     expand_office_description(office_sought),
-                    report_type
-                ], ' ')
+                    report_type,
+                    'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 6]
```diff
@@ -68,27 +69,29 @@ returns text as $$
...
+                ], ' '
             when array_length(election_state, 1) > 1 then array_to_string(
                 array[
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'Report Multi-state'
+                    'Report Multi-state due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 7]
```diff
@@ -68,27 +69,29 @@ returns text as $$
...
                 ], ' ')
             else
                 array_to_string(
                 array[
                     array_to_string(election_state, ', '),
                     expand_office_description(office_sought),
-                    rpt_tp_desc
+                    rpt_tp_desc,
+                    'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 8]
```diff
@@ -162,9 +165,8 @@ with elections_raw as(
         array_to_string(array[
             expand_office_description(office_sought::text),
             clean_report(rpt_tp_desc::text),
-            'Due. Report code:',
-            report_type::text,
-            array_to_string(array_agg(election_state order by election_state)::text[], ', ')
+            'due today',
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 9]
```diff
@@ -38,20 +38,28 @@ create or replace function generate_election_title(trc_election_type_id text, of
...
                 'Multi-state'::text,
-                ' due today'
+                'held today'
+            ], ' ')
+        when array_length(contest, 1) = 0 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'held today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 10]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
                     report_type,
-                    'Report Multi-state due today'
+                    'report multi-state due today'
+                ], ' ')
+            when pt_tp_desc is null and array_length(election_state, 1) = 0 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 11]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'Report Multi-state due today'
+                    'report multi-state due today'
+                ], ' ')
+            when array_length(election_state, 1) = 0 then array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    rpt_tp_desc,
+                    'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 12]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
                 ], ' ')
             else
                 array_to_string(
                 array[
-                    array_to_string(election_state, ', '),
+                    array_to_string(election_state, ', ') || ':',
                     expand_office_description(office_sought),
                     rpt_tp_desc,
                     'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 13]
```diff
@@ -38,20 +38,28 @@ create or replace function generate_election_title(trc_election_type_id text, of
...
-        when array_length(contest, 1) > 1 then array_to_string(
+        when array_length(contest, 1) > 3 then array_to_string(
             array[
                 party,
                 office_sought,
-                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
                 'Multi-state'::text,
-                ' due today'
+                'held today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 14]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
     begin
         return case
-            when rpt_tp_desc is null and array_length(election_state, 1) > 1 then
+            when rpt_tp_desc is null and array_length(election_state, 1) > 3 then
                 array_to_string(
                 array[
                     expand_office_description(office_sought),
                     report_type,
-                    'Report Multi-state due today'
+                    'report multi-state due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 15]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
+                    'due today'
                 ], ' ')
             when rpt_tp_desc is null then
                 array_to_string(
                 array[
-                    array_to_string(election_state, ', '),
+                    array_to_string(election_state, ', ') || ':',
                     expand_office_description(office_sought),
                     report_type,
                     'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 16]
```diff
@@ -64,31 +72,44 @@ create or replace function name_reports(office_sought text, report_type text, rp
...
                 ], ' '
-            when array_length(election_state, 1) > 1 then array_to_string(
+            when array_length(election_state, 1) > 3 then array_to_string(
                 array[
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'Report Multi-state due today'
+                    'report multi-state due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today'

File: data/sql_updates/omnibus_dates.sql [hunk 17]
```diff
@@ -165,8 +187,8 @@ with elections_raw as(
         array_to_string(array[
             expand_office_description(office_sought::text),
             clean_report(rpt_tp_desc::text),
-            'due today',
-            array_to_strding(array_agg(election_state order by election_state)::text[], ', ')
+            array_to_string(array_agg(election_state order by election_state)::text[], ', '),
+            'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 18]
```diff
@@ -89,11 +89,11 @@ returns text as $$
             when rpt_tp_desc is null then
                 array_to_string(
                 array[
-                    array_to_string(election_state, ', ') || ':',
+                    array_to_string(election_state, ', '),
                     expand_office_description(office_sought),
                     report_type,
                     'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 19]
```diff
@@ -79,7 +79,7 @@ returns text as $$
                     report_type,
                     'report multi-state due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 20]
```diff
@@ -109,7 +109,7 @@ returns text as $$
             else
                 array_to_string(
                 array[
-                    array_to_string(election_state, ', ') || ':',
+                    array_to_string(election_state, ', '),
                     expand_office_description(office_sought),
                     rpt_tp_desc,
                     'due today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 21]
```diff
@@ -44,22 +44,22 @@ returns text as $$
                 office_sought,
                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
                 'Multi-state'::text,
-                'held today'
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 22]
```diff
@@ -44,22 +44,22 @@ returns text as $$
...
             ], ' ')
         when array_length(contest, 1) = 0 then array_to_string(
             array[
                 party,
                 office_sought,
                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
-                'held today'
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 23]
```diff
@@ -44,22 +44,22 @@ returns text as $$
...
             ], ' ')
         else array_to_string(
             array[
                 array_to_string(contest, ', ') || ':',
                 party,
                 office_sought,
                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
-                'held today'
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 24]
```diff
@@ -77,42 +77,42 @@ returns text as $$
...
-                    'report multi-state due today'
+                    'Report Multi-state Due Today'
                 ], ' ')
             when rpt_tp_desc is null and array_length(election_state, 1) = 0 then
                 array_to_string(
                 array[
                     expand_office_description(office_sought),
                     report_type,
-                    'due today'
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 25]
```diff
@@ -77,42 +77,42 @@ returns text as $$
                 array[
                     expand_office_description(office_sought),
                     report_type,
-                    'report multi-state due today'
+                    'Report Multi-state Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 26]
```diff
@@ -77,42 +77,42 @@ returns text as $$
...
+                    'Report Due Today'
                 ], ' ')
             when rpt_tp_desc is null then
                 array_to_string(
                 array[
                     array_to_string(election_state, ', ') || ':',
                     expand_office_description(office_sought),
                     report_type,
-                    'due today'
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 27]
```diff
@@ -77,42 +77,42 @@ returns text as $$
...
                 ], ' ')
             when array_length(election_state, 1) > 3 then array_to_string(
                 array[
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'report multi-state due today'
+                    'Report Multi-state Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report (for Multiple States) is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 28]
```diff
@@ -77,42 +77,42 @@ returns text as $$
...
                 ], ' ')
             when array_length(election_state, 1) = 0 then array_to_string(
                 array[
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'due today'
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 29]
```diff
@@ -77,42 +77,42 @@ returns text as $$
...
                 ], ' ')
             else
                 array_to_string(
                 array[
                     array_to_string(election_state, ', ') || ':',
                     expand_office_description(office_sought),
                     rpt_tp_desc,
-                    'due today'
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 30]
```diff
@@ -152,7 +152,7 @@ with elections_raw as(
                 expand_office_description(office_sought::text),
                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
                 array_to_string(array_agg(contest order by contest)::text[], ', '),
-                'held today'
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 31]
```diff
@@ -188,7 +188,7 @@ with elections_raw as(
             expand_office_description(office_sought::text),
             clean_report(rpt_tp_desc::text),
             array_to_string(array_agg(election_state order by election_state)::text[], ', '),
-            'due today'
+            'Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 32]
```diff
@@ -66,13 +66,52 @@ returns text as $$
...
+create or replace function generate_election_summary(trc_election_type_id text, office_sought text, contest text[], party text, trc_election_id numeric)
+returns text as $$
+    begin
+        return case
+        when array_length(contest, 1) < 3 and array_length(contest, 1) >= 1 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today',
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 33]
```diff
@@ -66,13 +66,52 @@ returns text as $$
...
+                'States:',
+                array_to_string(contest, ', ')
+            ], ' ')
+        when array_length(contest, 1) = 0 then array_to_string(
+            array[
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 34]
```diff
@@ -66,13 +66,52 @@ returns text as $$
...
+            ], ' ')
+        else array_to_string(
+            array[
+                array_to_string(contest, ', ') || ':',
+                party,
+                office_sought,
+                expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
+                'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

File: data/sql_updates/omnibus_dates.sql [hunk 35]
```diff
@@ -94,7 +133,7 @@ returns text as $$
                     report_type,
                     'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 36]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+create or replace function generate_report_summary(office_sought text, report_type text, rpt_tp_desc text, election_state text[])
+returns text as $$
+    begin
+        return case
+            when rpt_tp_desc is null and array_length(election_state, 1) < 3 and array_length(election_state, 1) >= 1 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Due Today',
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 37]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+                    'States:',
+                    array_to_string(election_state, ', ')
+                ], ' ')
+            when rpt_tp_desc is null and array_length(election_state, 1) < 1 then
+                array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 38]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+                ], ' ')
+            when rpt_tp_desc is null then
+                array_to_string(
+                array[
+                    array_to_string(election_state, ', ') || ':',
+                    expand_office_description(office_sought),
+                    report_type,
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 39]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+                ], ' ')
+            when array_length(election_state, 1) < 3 and array_length(election_state, 1) >= 1 then array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    rpt_tp_desc,
+                    'Report Due Today',
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 40]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+                    'States:',
+                    array_to_string(election_state, ', ')
+                ], ' ')
+            when array_length(election_state, 1) = 0 then array_to_string(
+                array[
+                    expand_office_description(office_sought),
+                    rpt_tp_desc,
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 41]
```diff
@@ -119,6 +158,64 @@ returns text as $$
...
+                ], ' ')
+            else
+                array_to_string(
+                array[
+                    array_to_string(election_state, ', ') || ':',
+                    expand_office_description(office_sought),
+                    rpt_tp_desc,
+                    'Report Due Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: Report is Due Today

File: data/sql_updates/omnibus_dates.sql [hunk 42]
```diff
@@ -31,19 +31,19 @@ $$ language plpgsql;
...
         return case
-        when array_length(contest, 1) > 3 then array_to_string(
+        when array_length(contest, 1) < 3 and array_length(contest, 1) >= 1 then array_to_string(
             array[
                 party,
                 office_sought,
                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric),
-                'Multi-state'::text,
+                'Multi-state',
                 'Held Today'
```
Comment (emileighoutlaw) [inline data/sql_updates/omnibus_dates.sql]: is Held Today

**Comment by emileighoutlaw** [comment]
Yay! Thank you!! You are the best, and I am the worst (because I did a bad job of marking all the verbs I wanted, and I found a lot more. But I promise this is the last pass)

**Comment by LindsayYoung** [comment]
I already did the ones you marked ;)

**Comment by emileighoutlaw** [comment]
Ah! You are the best! Thank you! ❤️

File: data/functions/calendar.sql [hunk 43]
```diff
@@ -0,0 +1,223 @@
...
+$$ language plpgsql;
+
+
+-- Trying to make the names flow together as best as possible
+-- To keep the titles concise states are abbreviated as multi state if there is more than one
+-- Like:
+    -- FL: House General Election Held Today
+    -- NH, DE: DEM Convention Held Today
+    -- General Election Multi-state Held Today
+create or replace function generate_election_description(election_type text, office_sought text, contest text[], party text)
```
Comment (ccostino) [inline data/functions/calendar.sql]: This was renamed from `general_election_title`, correct?

File: data/sql_updates/omnibus_dates.sql [hunk 44]
```diff
@@ -1,100 +1,5 @@
...
-                 expand_election_type_caucus_convention_clean(trc_election_type_id::text, trc_election_id::numeric)
-            ], ' ')
-        end;
-    end
-$$ language plpgsql;
-
-
--- Not all report types are on dimreporttype, so for the reports to all have
--- titles, I am adding a case. Ideally, we would want the right mapping.
-create or replace function name_reports(office_sought text, report_type text, rpt_tp_desc text, election_state text[])
```
Comment (ccostino) [inline data/sql_updates/omnibus_dates.sql]: And this was renamed to one of the functions above?

File: data/sql_updates/omnibus_dates.sql [hunk 45]
```diff
@@ -108,28 +13,29 @@ with elections_raw as(
...
     from
         trc_election
     where
         trc_election_status_id = 1
 ), elections as (
     select
         'election'::text as category,
-        generate_election_title(
-            trc_election_type_id::text,
+        generate_election_description(
```
Comment (ccostino) [inline data/sql_updates/omnibus_dates.sql]: That'll teach me to ask questions too fast - I see the rename here. :-)

File: data/functions/calendar.sql (hunk 43 above)
Comment (ccostino) [inline data/functions/calendar.sql]: Disregard!

File: data/sql_updates/omnibus_dates.sql (hunk 44 above)
Comment (ccostino) [inline data/sql_updates/omnibus_dates.sql]: Disregard!

**Comment by ccostino** [comment]
Thank you @LindsayYoung and @emileighoutlaw!

**Pull request merged**